### DISCUSSION TITLE
RoadMap and Discussion of MMDet 3.x

### DISCUSSION THREAD

**Discussion by ZwwWayne** [open-mmlab/mmdetection#9318] [Ideas]
Hi, the developers

We are glad to release MMDet 3.x since Sept 1st, 2022, with so many exciting changes providing stronger and more general benchmarks and cutting edges with easier usage for object detection. You can find the details in the [initial release](https://github.com/open-mmlab/mmdetection/releases/tag/v3.0.0rc0).

As MMDet 3.x is brand new, we are still completing its functionalities and documentation and we welcome any kind of contribution including issues/PRs/discussions. Moreover, we want to hear more from you so that MMDet 3.x could suit your projects of any purpose. 
Therefore, we keep this discussion open to collect suggestions/ideas from users and hear your voice. Our monthly release plan will be updated in different [milestones](https://github.com/open-mmlab/mmdetection/milestones), and we will open discussion for each future release of MMDet 3.x, which will also be listed here. You can also leave Chinese message in https://github.com/open-mmlab/mmdetection/discussions/8779 if you prefer.

In this discussion, you can either:
1. Suggest a new feature/improvement by leaving a comment.
2. Vote for a feature request with :+1: or be against with :-1:. (Remember that developers are busy and cannot respond to all feature requests, so vote for your most favorable one!)
3. Tell us that you would like to help implement one of the features in the list or review the PRs. (This is the greatest things to hear about!)

**Comment by JosonChan1998** [comment]
Any plan for the DLA backbone? If needed, I can try to implement it with fcos :smile:.

**Reply by ZwwWayne** [reply]
It would be great if you would like to! We also suggest supporting it with CenterNet as the performance DLA34+CenterNet has a reference.

**Comment by IgnatovD** [comment]
Hello! Thanks a lot for your hard work!

I have my own dataset for 3 classes in COCO format. I trained the model on this data. Is there currently a way to use the trained model to get metrics on the test data (test data is labeled)?

I tried to run the script https://github.com/open-mmlab/mmdetection/blob/master/tools/analysis_tools/eval_metric.py but an error occurs, I think it is because it processes only the original COCO dataset for 80 classes.

```
Traceback (most recent call last):
  File "/home/jovyan/.cache/pypoetry/virtualenvs/samolet-fUQ9Ej7U-py3.8/lib/python3.8/site-packages/mmdet/.mim/tools/analysis_tools/eval_metric.py", line 88, in <module>
    main()
  File "/home/jovyan/.cache/pypoetry/virtualenvs/samolet-fUQ9Ej7U-py3.8/lib/python3.8/site-packages/mmdet/.mim/tools/analysis_tools/eval_metric.py", line 84, in main
    print(dataset.evaluate(outputs, **eval_kwargs))
  File "/home/jovyan/.cache/pypoetry/virtualenvs/samolet-fUQ9Ej7U-py3.8/lib/python3.8/site-packages/mmdet/datasets/coco.py", line 641, in evaluate
    result_files, tmp_dir = self.format_results(results, jsonfile_prefix)
  File "/home/jovyan/.cache/pypoetry/virtualenvs/samolet-fUQ9Ej7U-py3.8/lib/python3.8/site-packages/mmdet/datasets/coco.py", line 383, in format_results
    result_files = self.results2json(results, jsonfile_prefix)
  File "/home/jovyan/.cache/pypoetry/virtualenvs/samolet-fUQ9Ej7U-py3.8/lib/python3.8/site-packages/mmdet/datasets/coco.py", line 315, in results2json
    json_results = self._det2json(results)
  File "/home/jovyan/.cache/pypoetry/virtualenvs/samolet-fUQ9Ej7U-py3.8/lib/python3.8/site-packages/mmdet/datasets/coco.py", line 252, in _det2json
    data['category_id'] = self.cat_ids[label]
IndexError: list index out of range
```

But I managed to run this script https://github.com/open-mmlab/mmdetection/blob/master/tools/analysis_tools/analyze_results.py on my 3 classes by setting the parameter `data.test.classes='("class1", "class2", "class3")'`.

I've read all the documentation but haven't found a way to get quality scores on user test data.

**Reply by ZwwWayne** [reply]
This is not the issue of mmdet3.x. You need to also modify the number of classes in the head.

**Comment by shelper** [comment]
great job, but i would like to see the support for tflite so it can be easily deployed to android devices.
i think to do this, there is a need to make the post processing part in the model (everything after the loss calculation, i believe) removable, so the model has better compatibility with TFLITE, and can be readily converted to tensorflow -> tflite, through onnx or openvino.

I can contribute to the code for such conversion, 'cus i went through quite a few issues converting the .pth model to tflite try to see if i can use the model on android .

**Comment by zz4fap** [comment]
Inference on video streams. Just the same YOLOv8 does.
